signer.go

```diff
@@ -23,6 +23,16 @@ type Signer interface {
 	Sign(rand io.Reader, content []byte) ([]byte, error)
 }
 
+// DigestSigner is an interface for private keys to sign digested COSE signatures.
+type DigestSigner interface {
+	Signer
```

Comment:
Does `DigestSigner` require `Sign(rand io.Reader, content []byte) ([]byte, error)`?

If no, it should be replaced with
```suggestion
	// Algorithm returns the signing algorithm associated with the public key.
	Algorithm() Algorithm
```

Reply:
Not really, good catch.